src/h5web/vis-packs/core/scalar/ScalarVis.tsx

```diff
@@ -8,7 +9,12 @@ interface Props {
 
 function ScalarVis(props: Props) {
   const { value } = props;
-  return <div className={styles.scalar}>{value.toString()}</div>;
+
+  return (
+    <div className={styles.scalar}>
+      {Array.isArray(value) ? formatComplexValue(value) : value.toString()}
```

Comment:
The check leaves a lot to be desired in my opinion. It is sufficient as `value` is `string | number | boolean | [number, number]` but could we do better ?

Reply:
Same remark as with `MatrixVis`. We could have a generic on `ScalarVis` and `MatrixVis`:

```ts
interface Props<T> {
  value: T;
  formatter: (value: T) => string;
}
```

... and a utility function that takes a dataset (or attribute, so it can also be used in the metadata viewer?) and returns a formatter with a conditional type similar to `Value`:

```ts
function getFormatter<T extends Dataset | Attribute>(datasetOrAttr: T): Formatter<T> { ... }
```